### PR TITLE
Fix: output being truncated and watcher crash

### PR DESCRIPTION
**Pull request by SRWieZ**
Hello friends 👋!

TLDR : Fix Exception when too much files are updated at once.

## Context
I was using this library to watch a Laravel project. When I did a `composer update` in the watched directory the script crashed with the following error: **Undefined array key 1**  referring to [this line](https://github.com/spatie/file-system-watcher/blob/48ada9f48c2474823d6f8653f77612ef89f3f5f4/src/Watch.php#L169C1-L169C1)
Tried a few times and every time a batch of files is updated at once I get the same error again.

With the help of `ray()` to see what was happening, I found the culprit.
```php
            try {
                [$type, $path] = explode(' - ', $line, 2);
            } catch (\Exception $e) {
                ray($line);
                continue;
            }
```

Here is a screenshot cleary showing what's wrong
Capture d’écran 2023-10-29 à 10 51 33

## The fix: what did I changed?
1. $buffer accumulates the incremental output. 
2. strrpos() function locates the last occurrence of PHP_EOL. 
3. If found, it indicates that at least one complete line exists in the $buffer. 
4. We then extract this complete lines and update the $buffer accordingly. 
5. Then explode these complete lines into an array $array and proceed with actOnOutput() function.

You will also notice I changed some lines on the tests but this is purely because my computer is slower than yours and I could not test the package even before my change!

Tests before and after my PR, same time, same result.
Capture d’écran 2023-10-29 à 11 17 58

And, thanks for all your packages guys !

### CODE REVIEW

**Comment by SRWieZ** [comment]
To complement my research : I wondered if Laravel Octane had the same problem since it use Chokidar too. The main difference is Octane just watch for changes and restart workers. Here we use the output to know which files has been updated and this output can become large enough to be slice between getIncrementalOutput calls.
